assembly/src/assembler/module_provider.rs

```diff
@@ -0,0 +1,59 @@
+use super::{BTreeMap, Library, LibraryError, Module, ProcedureId, Vec};
+
+// MODULE PROVIDER
+// ================================================================================================
+
+/// A module provider that will store the procedure ASTs used during the compilation process.
```

Comment:
nit: I would change this to something like: "A module provider contains all modules from libraries available to a given assembler. It is used during compilation to resolve references to imported procedures."

Reply:
Fixed!

Reply:
My bad; fixed!